doc/user/content/transform-data/patterns/percentiles.md

```diff
@@ -10,134 +10,306 @@ menu:
 
-The cumulative count and the cumulative distribution can then be used to query for arbitrary percentiles. The following query returns the 90-th percentile.
+HDR histograms are related to how [floating point numbers are
+represented](https://en.wikipedia.org/wiki/Double-precision_floating-point_format)
+as integers. The underlying assumption is that smaller numbers require a higher
+precision to be distinguishable (e.g. 5 ms and 6 ms are different and should be
+in different buckets) whereas larger numbers can be rounded more aggressively as
+their relative error becomes less relevant (e.g. 10000 ms and 10001 ms are
+almost the same and can reside in the same bucket).
 
-```mzsql
-SELECT bucket AS percentile
-FROM distribution
-WHERE cumulative_distribution >= 0.9
-ORDER BY cumulative_distribution
-LIMIT 1;
-```
+In the example below, to reduce the number of buckets, the values are first
+decomposed into their floating point representation (i.e., `mantissa *
+2^exponent`), and then with the precision of the mantissa lowered,
+reconstructed for the respective bucket value.
 
-To increase query performance, it can make sense to keep the `distribution` always up to date by creating an index on the view:
+- With higher precisions, fewer items are kept in the same bucket and thus, more
+  memory is required, but the approximate percentile becomes more precise.
 
-```mzsql
-CREATE INDEX distribution_idx ON distribution (cumulative_distribution);
-```
+- With lower precisions, more items are kept in the same bucket, and thus, the
+  less memory is required, but the approximate percentile becomes less precise.
 
-Histograms work well for a domain with low cardinality. But note the cross join in the definition of `distribution`. As a consequence, the required memory grows quadratic in the number of unique values and may therefore grow large if the domain has many unique values.
+While the basic ideas of [using histograms to compute
+percentiles](#using-histograms-to-compute-percentiles) remains the same for HDR
+histograms, the bucket calculation is more involved as each bucket is now
+represented by the tuple (mantissa, exponent).
 
+### Example
 
-## Using HDR histograms to compute approximate percentiles
+{{< tip >}}
 
-[HDR histograms](https://github.com/HdrHistogram/HdrHistogram) can be used to approximate percentiles in a space efficient manner that scales well even for large domains with many distinct values. HDR histograms reduce the precision of values that are tracked and use buckets with variable width. Buckets that are closer to 0 are smaller whereas buckets far away from 0 are wider. This works particularly well for data that exhibits a long tail of large values, e.g., latency measurements.
+The following example assumes you have not previously created and populated the
+`input` table from the [Using histograms to compute percentiles
+example](#example). If you have created and populated the table, skip to step 3.
```

Comment:
Maybe fine, but I can see the "step 3" reference rotting if we change the steps ever. Alternately, maybe: "skip the corresponding steps".

Reply:
Updated in patch 75f32afbf3839b0dc7b5f84321847c6c88161bff